scripts/build_latest_xorg_apt.sh

```diff
@@ -0,0 +1,231 @@
+#!/bin/sh
+################################################################################
+# B U I L D   W I T H   L A T E S T   X O R G   A P T . S H
+#
+# Builds the latest version of the xserver from gitlab.freedesktop.org
+#
+# (c) Matt Burt  2025
+#
+# Usage ./build_with_latest_xorg_apt.sh [-r] <build_dir>
+#
+# Add [-r] to use an autoresume file. This can be useful when tracking down
+# problems with the build script. It can also make things very confusing!
+#
+# sudo is used to install build dependencies, plus a few package
+# dependencies.
+#
+# Most dependencies are pulled down and installed as the build user.
+################################################################################
+
+# ------------------------------------------------------------------------------
+# G L O B A L S
+# ------------------------------------------------------------------------------
+# Build tools
+APT_PKG_LIST="git pip pkg-config cmake automake autoconf libtool"
+APT_PKG_LIST="$APT_PKG_LIST python3-venv" ; # meson/ninja
+# Other OS-provided dependencies
+#
+# We pull in most of Mesa rather than building it from scratch as
+# Mesa dependencies (notably the rust compiler version) can be
+# problematic
+APT_PKG_LIST="$APT_PKG_LIST mesa-common-dev" ; # See above
+APT_PKG_LIST="$APT_PKG_LIST libfreetype-dev" ; # Needed by libXfont
+APT_PKG_LIST="$APT_PKG_LIST libudev-dev" ; # Needed by xserver
+APT_PKG_LIST="$APT_PKG_LIST libssl-dev" ; # Needed by xserver
+APT_PKG_LIST="$APT_PKG_LIST libepoxy-dev" ; # Needed by xserver
+
+# We use 'pip' to get the latest meson and ninja build tools down. These
+# are installed for the build user.
+PIP_PKG_LIST="meson ninja"
+
+# The modular packages supported by the xserver build tool which we need to
+# build. This includes the xserver package
+#
+# This list of packages is sorted into dependency order before use.
+MODULAR_PKG_LIST="util/macros"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST font/util"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST proto/xorgproto proto/xcbproto"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST lib/libxcvt lib/libxtrans lib/libXau"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST lib/libXdmcp lib/libxcb lib/libX11"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST lib/libfontenc lib/libXfont lib/libxkbfile"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST lib/libxshmfence"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST mesa/drm"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST pixman/pixman"
+MODULAR_PKG_LIST="$MODULAR_PKG_LIST xserver"
+
+# Check the XDG_RUNTIME_DIR is specified, and use it for the autoresume file
+if [ -z "$XDG_RUNTIME_DIR" ]; then
+    export XDG_RUNTIME_DIR=/run/user/$(id -u)
+fi
+AUTORESUME_FILE="$XDG_RUNTIME_DIR/xserver-autores.txt"
+
+# Our source dir. This is currently hard-coded
+SOURCE_DIR="$HOME/xserver-src"
+
+# Dependencies in the source dir
+BUILDER="$SOURCE_DIR/util/modular/build.sh" ; # xorg modular build script
```

Comment:
Where is this script coming from ?

Reply:
It's fetched here:-

https://github.com/neutrinolabs/xorgxrdp/blob/4e9b011e9bcff8affb77e65bb6a9e8cdc92cde1b/scripts/build_latest_xorg_apt.sh#L210-L213

This could be clearer though. I'll update the comment and add an additional check after the checkout.